src/library/dao/playlistdao.cpp

```diff
@@ -246,6 +246,29 @@ bool PlaylistDAO::isPlaylistLocked(const int playlistId) const {
     return false;
 }
 
+bool PlaylistDAO::clearPlaylist(const int playlistId) {
+    int position =
+        (m_pAutoDJProcessor && m_pAutoDJProcessor->nextTrackLoaded()) ? 2 : 1;
+    ScopedTransaction transaction(m_database);
+    QSqlQuery query(m_database);
+    query.prepare("DELETE FROM PlaylistTracks "
+                  "WHERE playlist_id=:id AND position>=:pos");
+    query.bindValue(":id", playlistId);
+    query.bindValue(":pos", position);
+    if (!query.exec()) {
+        LOG_FAILED_QUERY(query);
+        return false;
+    }
+    transaction.commit();
+    emit(changed(playlistId));
+    return true;
+}
+
+bool PlaylistDAO::replaceTracksInPlaylist(const QList<TrackId>& trackIds, const int playlistId) {
+    return clearPlaylist(playlistId)
```

Comment:
Since this is so short, I would write it directly in sendToAutoDJ instead of adding a method.